app/assets/javascripts/initializers/initScrolling.js.erb

```diff
@@ -24,21 +24,38 @@ function checkIfNearBottomOfPage() {
 }
 
 function fetchNextPageIfNearBottom() {
-  var elCheck = document.getElementById("index-container") && !document.getElementById("query-wrapper");
+  var indexContainer = document.getElementById("index-container");
+  var elCheck =  indexContainer && !document.getElementById("query-wrapper");
   if (!elCheck) {
     return;
   }
-  if ( !done && !fetching && (window.scrollY) > (document.documentElement.scrollHeight - 3700) ) {
+  
+  var indexWhich = indexContainer.dataset.which;
+
+  var fetchCallback;
+  var scrollableElemId;
+  if (indexWhich == "podcast-episodes") {
```

Comment:
I refactored some of the dataset-specific logic here.

As far as I could tell, `videos` are the only dataset that use something other than `articles-list` as the scrollable element. Please check me on this.

I also slightly refactored the fetching logic into callbacks here, because it felt unnecessary to do additional conditional checks on the dataset to figure out which fetching mechanism we should use.